ui/packages/catalog-ui-search/src/main/webapp/react-component/container/metacard-interactions/index.tsx

```diff
@@ -0,0 +1,405 @@
+      const choice = confirmation.get('choice')
+      if (choice === true) {
+        const loadingView = new LoadingView()
+        store.get('workspaces').once('sync', (workspace: any) => {
+          loadingView.remove()
+          wreqr.vent.trigger('router:navigate', {
+            fragment: `workspaces/${workspace.id}`,
+            options: {
+              trigger: true,
+            },
+          })
+        })
+        store.get('workspaces').createWorkspaceWithQuery(newQuery)
+      } else if (choice !== false) {
+        store.getCurrentQueries().remove(choice)
+        store.getCurrentQueries().add(newQuery)
+        store.setCurrentQuery(newQuery)
+      }
+    }
+  )
+}
+
+const handleShow = (context: Props) => {
+  const preferences = user.get('user').get('preferences')
+  const getResult = (result: Result) =>
+    result
+      .get('metacard')
+      .get('properties')
+      .get('id')
+
+  preferences.get('resultBlacklist').remove(context.model.map(getResult))
+  preferences.savePreferences()
+}
+
+const handleHide = (context: Props) => {
+  const preferences = user.get('user').get('preferences')
+  const getResult = (result: Result) => ({
+    id: result
+      .get('metacard')
+      .get('properties')
+      .get('id'),
+    title: result
+      .get('metacard')
+      .get('properties')
+      .get('title'),
+  })
+
+  preferences.get('resultBlacklist').add(context.model.map(getResult))
+  preferences.savePreferences()
+}
+
+const handleExpand = (context: Props) => {
+  const id = context.model
+    .first()
+    .get('metacard')
+    .get('properties')
+    .get('id')
+
+  wreqr.vent.trigger('router:navigate', {
+    fragment: 'metacards/' + id,
+    options: {
+      trigger: true,
+    },
+  })
+}
+
+const handleShare = (context: Props) =>
+  context.handleShare && context.handleShare()
+
+const appendCssIfNeeded = (model: Model, el: El) => {
+  const isMixed = model.reduce((accum: number, item: Result) => {
+    if (item.isRemote()) {
+      el.toggleClass('is-remote', true)
+      return ++accum
+    }
+    if (item.isWorkspace()) {
+      el.toggleClass('is-workspace', true)
+      return ++accum
+    }
+    if (item.isResource()) {
+      el.toggleClass('is-resource', true)
+      return ++accum
+    }
+    if (item.isRevision()) {
+      el.toggleClass('is-revision', true)
+      return ++accum
+    }
+    if (item.isDeleted()) {
+      el.toggleClass('is-deleted', true)
+      return ++accum
+    }
+  }, 0)
+
+  el.toggleClass('is-mixed', isMixed > 1)
+
+  const currentWorkspace = store.getCurrentWorkspace()
+  el.toggleClass('in-workspace', Boolean(currentWorkspace))
+  el.toggleClass('is-downloadable', isDownloadable(model))
+  el.toggleClass('is-multiple', model.length > 1)
+  el.toggleClass('is-routed', isRouted())
+  el.toggleClass('is-blacklisted', isBlacklisted(model))
+  el.toggleClass('has-location', hasLocation(model))
+}
+
+const isDownloadable = (model: Model) =>
+  model.find((result: Result) =>
```

Comment:
I think this change got left out.  As of now we'll still get undefined or a model rather than a boolean.  

If we change `find` to `some` I think we'll get the behavior we want.